plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -343,11 +343,11 @@ In Haskell this is conditionally error, in PLC it is unconditionally error.
 
 These sorts of thing can be written by the user, or generated by GHC.
 
-We solve this by compiling such let-bindings *non-strictly*. That means we delay the body
-and force all uses of it. This is a bit overenthusiastic, as it will add overhead to
-effect-free let-bindings, but I don't know of an easy way to tell if an expression is pure.
+We solve this by compiling let-bindings *non-strictly*. That means we delay the body
+and force all uses of it.
 
 Conveniently, we can just use PIR's support for non-strict let bindings to implement this.
+The PIR optimizer will also strictify any such bindings that turn out to be pure.
```

Comment:
I don't know if we use the term "PIR optimizer" anywhere else, so the term seems a bit confusing. In any case, it would be good to say that:  "The PIR optimizer (which is enabled by default) ..."